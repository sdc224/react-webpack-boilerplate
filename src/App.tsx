import React from "react";

export default function App(): JSX.Element {
	return <div>Hello World</div>;
}
