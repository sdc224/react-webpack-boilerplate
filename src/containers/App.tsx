import React from "react";

interface AppProps {
	children?: React.ReactNode;
}

export default function App({ children }: AppProps) {
	return <>{children}</>;
}
