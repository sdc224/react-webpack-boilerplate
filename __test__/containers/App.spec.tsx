import React from "react";
import { render, RenderResult } from "@testing-library/react";
import App from "@containers/App";
import MockComponent from "../mocks/MockComponent";

let documentBody: RenderResult;

describe("<App />", () => {
	beforeEach(() => {
		documentBody = render(
			<App>
				<MockComponent />
			</App>
		);
	});
	it("has mock test", () => {
		const { getByText } = documentBody;
		expect(getByText(/Mock Component/i)).toBeInTheDocument();
	});
	it("matches snapshot", () => {
		const { baseElement } = documentBody;
		expect(baseElement).toMatchSnapshot();
	});
});
