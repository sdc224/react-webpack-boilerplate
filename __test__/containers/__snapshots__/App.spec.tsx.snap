// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<body>
  <div>
    <p>
      Mock Component
    </p>
  </div>
</body>
`;
